h/h_api/exceptions.py

```diff
@@ -0,0 +1,67 @@
+from h.h_api.model.json_api import JSONAPIError, JSONAPIErrorBody
+
+
+class JSONAPIException(Exception):
```

Comment:
This will form the basis for all our exceptions, so they can be trivial written out as JSON